chips/imxrt1050/src/gpio.rs

```diff
@@ -0,0 +1,1119 @@
+            }
+            _ => {}
+        }
+    }
+
+    fn set_output_low(&self) {
+        let port = self.pinid.get_port();
+
+        match self.pinid.get_pin_number() {
+            0b01001 => port.registers.dr.write(DR::DR9::CLEAR),
+            _ => {}
+        }
+    }
+
+    fn is_output_high(&self) -> bool {
+        let port = self.pinid.get_port();
+
+        match self.pinid.get_pin_number() {
+            0b01001 => port.registers.dr.is_set(DR::DR9),
+            _ => false,
+        }
+    }
+
+    fn toggle_output(&self) -> bool {
+        if self.is_output_high() {
+            self.set_output_low();
+            false
+        } else {
+            self.set_output_high();
+            true
+        }
+    }
+
+    fn read_input(&self) -> bool {
+        let port = self.pinid.get_port();
+
+        match self.pinid.get_pin_number() {
+            0b1001 => port.registers.dr.is_set(DR::DR9),
+            _ => false,
+        }
+    }
+}
+
+impl hil::gpio::Pin for Pin<'_> {}
+impl<'a> hil::gpio::InterruptPin<'a> for Pin<'a> {}
+
+impl hil::gpio::Configure for Pin<'_> {
+    fn make_output(&self) -> hil::gpio::Configuration {
+        self.set_mode(Mode::Output);
+        hil::gpio::Configuration::Output
+    }
+
+    fn make_input(&self) -> hil::gpio::Configuration {
+        self.set_mode(Mode::Input);
+        hil::gpio::Configuration::Input
+    }
+
+    fn deactivate_to_low_power(&self) {
+        // Not implemented yet
+    }
+
+    fn disable_output(&self) -> hil::gpio::Configuration {
+        // Not implemented yet
+        hil::gpio::Configuration::LowPower
+    }
+
+    fn disable_input(&self) -> hil::gpio::Configuration {
+        // Not implemented yet
+        hil::gpio::Configuration::LowPower
+    }
+
+    // PullUp or PullDown mode are set through the Iomux module
+    fn set_floating_state(&self, _mode: hil::gpio::FloatingState) {}
+
+    fn floating_state(&self) -> hil::gpio::FloatingState {
+        hil::gpio::FloatingState::PullNone
+    }
+
+    fn configuration(&self) -> hil::gpio::Configuration {
+        match self.get_mode() {
+            Mode::Input => hil::gpio::Configuration::Input,
+            Mode::Output => hil::gpio::Configuration::Output,
+        }
+    }
+
+    fn is_input(&self) -> bool {
+        self.get_mode() == Mode::Input
+    }
+
+    fn is_output(&self) -> bool {
+        self.get_mode() == Mode::Output
+    }
+}
+
+impl hil::gpio::Output for Pin<'_> {
+    fn set(&self) {
+        self.set_output_high();
+    }
+
+    fn clear(&self) {
+        self.set_output_low();
+    }
+
+    fn toggle(&self) -> bool {
+        self.toggle_output()
+    }
+}
+
+impl hil::gpio::Input for Pin<'_> {
+    fn read(&self) -> bool {
+        self.read_input()
+    }
+}
+
+/// Interrupt capabilities are not yet implemented
+impl<'a> hil::gpio::Interrupt<'a> for Pin<'a> {
+    fn enable_interrupts(&self, _mode: hil::gpio::InterruptEdge) {}
+
+    fn disable_interrupts(&self) {}
```

Comment:
The porting guide (https://github.com/tock/tock/blob/master/doc/Porting.md) specifies that GPIO with interrupts is a prerequisite for a chip to be added to Tock